### PR TITLE
samples: mpsl: timeslot: Fix missing empty_app_core in build

### PR DESCRIPTION
**Pull request by nordicjm**
Adds the empty_app_core application to the build so that it works out of the box on the nrf53 without needing a separate image built and flashed

### CODE REVIEW

File: samples/mpsl/timeslot/sysbuild.cmake [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  # Get application core board target
+  string(REPLACE "/" ";" split_board_qualifiers "${BOARD_QUALIFIERS}")
+  list(GET split_board_qualifiers 1 target_soc)
+  list(GET split_board_qualifiers 2 target_cpucluster)
+
+  if(DEFINED BOARD_REVISION)
+    set(board_target_appcore "${BOARD}@${BOARD_REVISION}/${target_soc}/cpuapp")
+  else()
+    set(board_target_appcore "${BOARD}/${target_soc}/cpuapp")
+  endif()
```
Comment (tejlmand) [inline samples/mpsl/timeslot/sysbuild.cmake:19]: Revision should be passed independently.
```suggestion
  set(board_target_appcore "${BOARD}/${target_soc}/cpuapp")
```

File: samples/mpsl/timeslot/sysbuild.cmake [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+  set(target_soc)
+  set(target_cpucluster)
+
+  ExternalZephyrProject_Add(
+    APPLICATION empty_app_core
+    SOURCE_DIR ${ZEPHYR_NRF_MODULE_DIR}/samples/nrf5340/empty_app_core
+    BOARD ${board_target_appcore}
+  )
```
Comment (tejlmand) [inline samples/mpsl/timeslot/sysbuild.cmake:28]: pass revision independently:
```suggestion
    BOARD_REVISION ${BOARD_REVISION}
  )
```

**Review by tejlmand — CHANGES_REQUESTED** [review]
minor adjustements

**Comment by hermabe** [comment]
It would be really nice if this was handled in the common cmake so that we didnt have to add all of this extra logic to every sample running on the 53 netcore. I already had to this to one other sample (https://github.com/nrfconnect/sdk-nrf/pull/15623), and to our internal ble tests. When we had child images this was just a single config (`CONFIG_NCS_SAMPLE_EMPTY_APP_CORE_CHILD_IMAGE`).

**Comment by tejlmand** [comment]
> It would be really nice if this was handled in the common cmake so that we didnt have to add all of this extra logic to every sample running on the 53 netcore.

Agree, but current proposal is acceptable for now, and can be improved to more generic approach afterwards.

File: samples/mpsl/timeslot/sysbuild.cmake (hunk 1 above)
Comment (nordicjm) [reply samples/mpsl/timeslot/sysbuild.cmake]: Fixed

File: samples/mpsl/timeslot/sysbuild.cmake (hunk 2 above)
Comment (nordicjm) [reply samples/mpsl/timeslot/sysbuild.cmake:23]: Fixed

**Comment by nordicjm** [comment]
@bencefr this should now support copying samples out of tree for customer boards

File: samples/mpsl/timeslot/sysbuild.cmake [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+
+get_property(PM_DOMAINS GLOBAL PROPERTY PM_DOMAINS)
+
+# Include app core image if enabled
+if(SB_CONFIG_SOC_NRF5340_CPUNET)
+  # Get application core board target
+  string(REPLACE "/" ";" split_board_qualifiers "${BOARD_QUALIFIERS}")
+  list(GET split_board_qualifiers 1 target_soc)
+  list(GET split_board_qualifiers 2 target_cpucluster)
```
Comment (tejlmand) [inline samples/mpsl/timeslot/sysbuild.cmake:13]: Seems like dead code not being used below.
```suggestion
```

File: samples/mpsl/timeslot/sysbuild.cmake [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  set(board_target_appcore "${BOARD}/${target_soc}/cpuapp")
+  set(target_soc)
+  set(target_cpucluster)
```
Comment (tejlmand) [inline samples/mpsl/timeslot/sysbuild.cmake:16]: Can be removed together with above.
```suggestion
```

File: samples/esb/esb_prx/sysbuild.cmake [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+
+get_property(PM_DOMAINS GLOBAL PROPERTY PM_DOMAINS)
+
+# Include app core image if enabled
+if(SB_CONFIG_SOC_NRF5340_CPUNET)
+  # Get application core board target
+  string(REPLACE "/" ";" split_board_qualifiers "${BOARD_QUALIFIERS}")
+  list(GET split_board_qualifiers 1 target_soc)
+  list(GET split_board_qualifiers 2 target_cpucluster)
```
Comment (tejlmand) [inline samples/esb/esb_prx/sysbuild.cmake:13]:
```suggestion
```

File: samples/esb/esb_prx/sysbuild.cmake [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+  set(board_target_appcore "${BOARD}/${target_soc}/cpuapp")
+  set(target_soc)
+  set(target_cpucluster)
```
Comment (tejlmand) [inline samples/esb/esb_prx/sysbuild.cmake:16]:
```suggestion
```

File: samples/esb/esb_ptx/sysbuild.cmake [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+
+get_property(PM_DOMAINS GLOBAL PROPERTY PM_DOMAINS)
+
+# Include app core image if enabled
+if(SB_CONFIG_SOC_NRF5340_CPUNET)
+  # Get application core board target
+  string(REPLACE "/" ";" split_board_qualifiers "${BOARD_QUALIFIERS}")
+  list(GET split_board_qualifiers 1 target_soc)
+  list(GET split_board_qualifiers 2 target_cpucluster)
```
Comment (tejlmand) [inline samples/esb/esb_ptx/sysbuild.cmake:13]:
```suggestion
```

File: samples/esb/esb_ptx/sysbuild.cmake [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+  set(board_target_appcore "${BOARD}/${target_soc}/cpuapp")
+  set(target_soc)
+  set(target_cpucluster)
```
Comment (tejlmand) [inline samples/esb/esb_ptx/sysbuild.cmake:16]:
```suggestion
```

File: samples/bluetooth/connection_event_trigger/sysbuild.cmake [hunk 9]
```diff
@@ -2,12 +2,33 @@
...
 
-if (${SB_CONFIG_BOARD_NRF5340DK_NRF5340_CPUNET})
+get_property(PM_DOMAINS GLOBAL PROPERTY PM_DOMAINS)
+
+# Include app core image if enabled
+if(SB_CONFIG_SOC_NRF5340_CPUNET)
+  # Get application core board target
+  string(REPLACE "/" ";" split_board_qualifiers "${BOARD_QUALIFIERS}")
+  list(GET split_board_qualifiers 1 target_soc)
+  list(GET split_board_qualifiers 2 target_cpucluster)
```
Comment (tejlmand) [inline samples/bluetooth/connection_event_trigger/sysbuild.cmake:13]:
```suggestion
```

File: samples/bluetooth/connection_event_trigger/sysbuild.cmake [hunk 10]
```diff
@@ -2,12 +2,33 @@
...
+  set(board_target_appcore "${BOARD}/${target_soc}/cpuapp")
+  set(target_soc)
+  set(target_cpucluster)
```
Comment (tejlmand) [inline samples/bluetooth/connection_event_trigger/sysbuild.cmake:16]:
```suggestion
```

File: samples/bluetooth/direct_test_mode/sysbuild.cmake [hunk 11]
```diff
@@ -6,11 +6,20 @@
 get_property(PM_DOMAINS GLOBAL PROPERTY PM_DOMAINS)
 
 # Include app core image if enabled
-if(SB_CONFIG_SUPPORT_APPCORE AND DEFINED SB_CONFIG_APPCORE_IMAGE_NAME)
+if(SB_CONFIG_SOC_NRF5340_CPUNET AND DEFINED SB_CONFIG_APPCORE_IMAGE_NAME)
+  # Get application core board target
+  string(REPLACE "/" ";" split_board_qualifiers "${BOARD_QUALIFIERS}")
+  list(GET split_board_qualifiers 1 target_soc)
+  list(GET split_board_qualifiers 2 target_cpucluster)
```
Comment (tejlmand) [inline samples/bluetooth/direct_test_mode/sysbuild.cmake:13]:
```suggestion
```

File: samples/bluetooth/direct_test_mode/sysbuild.cmake [hunk 12]
```diff
@@ -6,11 +6,20 @@
...
+  set(board_target_appcore "${BOARD}/${target_soc}/cpuapp")
+  set(target_soc)
+  set(target_cpucluster)
```
Comment (tejlmand) [inline samples/bluetooth/direct_test_mode/sysbuild.cmake:16]:
```suggestion
```

File: samples/peripheral/802154_phy_test/sysbuild.cmake [hunk 13]
```diff
@@ -9,11 +9,20 @@ zephyr_get(EXTRA_ZEPHYR_MODULES)
 set(EXTRA_ZEPHYR_MODULES "${EXTRA_ZEPHYR_MODULES};${CMAKE_CURRENT_LIST_DIR}/modules/app_rpc" CACHE INTERNAL "extra modules directories")
 
 # Include app core image if enabled
-if(SB_CONFIG_BOARD_NRF5340DK_NRF5340_CPUNET)
+if(SB_CONFIG_SOC_NRF5340_CPUNET)
+  # Get application core board target
+  string(REPLACE "/" ";" split_board_qualifiers "${BOARD_QUALIFIERS}")
+  list(GET split_board_qualifiers 1 target_soc)
+  list(GET split_board_qualifiers 2 target_cpucluster)
```
Comment (tejlmand) [inline samples/peripheral/802154_phy_test/sysbuild.cmake:16]:
```suggestion
```

File: samples/peripheral/802154_phy_test/sysbuild.cmake [hunk 14]
```diff
@@ -9,11 +9,20 @@ zephyr_get(EXTRA_ZEPHYR_MODULES)
...
+  set(board_target_appcore "${BOARD}/${target_soc}/cpuapp")
+  set(target_soc)
+  set(target_cpucluster)
```
Comment (tejlmand) [inline samples/peripheral/802154_phy_test/sysbuild.cmake:19]:
```suggestion
```

File: samples/peripheral/radio_test/sysbuild.cmake [hunk 15]
```diff
@@ -6,11 +6,20 @@
...
 # Include app core image if enabled
-if(SB_CONFIG_BOARD_NRF5340DK_NRF5340_CPUNET)
+if(SB_CONFIG_SOC_NRF5340_CPUNET)
+  # Get application core board target
+  string(REPLACE "/" ";" split_board_qualifiers "${BOARD_QUALIFIERS}")
+  list(GET split_board_qualifiers 1 target_soc)
+  list(GET split_board_qualifiers 2 target_cpucluster)
+  set(board_target_appcore "${BOARD}/${target_soc}/cpuapp")
+  set(target_soc)
+  set(target_cpucluster)
```
Comment (tejlmand) [inline samples/peripheral/radio_test/sysbuild.cmake:16]:
```suggestion
```

File: samples/peripheral/radio_test/sysbuild.cmake [hunk 16]
```diff
@@ -6,11 +6,20 @@
 get_property(PM_DOMAINS GLOBAL PROPERTY PM_DOMAINS)
 
 # Include app core image if enabled
-if(SB_CONFIG_BOARD_NRF5340DK_NRF5340_CPUNET)
+if(SB_CONFIG_SOC_NRF5340_CPUNET)
+  # Get application core board target
+  string(REPLACE "/" ";" split_board_qualifiers "${BOARD_QUALIFIERS}")
+  list(GET split_board_qualifiers 1 target_soc)
+  list(GET split_board_qualifiers 2 target_cpucluster)
```
Comment (tejlmand) [inline samples/peripheral/radio_test/sysbuild.cmake:13]:
```suggestion
```

File: samples/nrf_rpc/entropy_nrf53/cpunet/sysbuild.cmake [hunk 17]
```diff
@@ -0,0 +1,34 @@
...
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+
+get_property(PM_DOMAINS GLOBAL PROPERTY PM_DOMAINS)
+
+# Include app core image if enabled
+if(SB_CONFIG_SOC_NRF5340_CPUNET)
+  # Get application core board target
+  string(REPLACE "/" ";" split_board_qualifiers "${BOARD_QUALIFIERS}")
+  list(GET split_board_qualifiers 1 target_soc)
+  list(GET split_board_qualifiers 2 target_cpucluster)
```
Comment (tejlmand) [inline samples/nrf_rpc/entropy_nrf53/cpunet/sysbuild.cmake:13]:
```suggestion
```

File: samples/nrf_rpc/entropy_nrf53/cpunet/sysbuild.cmake [hunk 18]
```diff
@@ -0,0 +1,34 @@
...
+  set(board_target_appcore "${BOARD}/${target_soc}/cpuapp")
+  set(target_soc)
+  set(target_cpucluster)
```
Comment (tejlmand) [inline samples/nrf_rpc/entropy_nrf53/cpunet/sysbuild.cmake:16]:
```suggestion
```

**Review by tejlmand — APPROVED** [review]
lgtm, but some dead code observed which can be removed, but not blocking.

File: samples/mpsl/timeslot/sysbuild.cmake (hunk 3 above)
Comment (nordicjm) [reply samples/mpsl/timeslot/sysbuild.cmake]: Fixed in all files

File: samples/nrf_rpc/entropy_nrf53/cpunet/sysbuild.cmake [hunk 19]
```diff
@@ -0,0 +1,32 @@
+#
+# Copyright (c) 2024 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+
+get_property(PM_DOMAINS GLOBAL PROPERTY PM_DOMAINS)
+
+# Include app core image if enabled
+if(SB_CONFIG_SOC_NRF5340_CPUNET)
```
Comment (KAGA164) [inline samples/nrf_rpc/entropy_nrf53/cpunet/sysbuild.cmake:9]: Empty app_core is not a target for app core in this sample

Comment (nordicjm) [reply samples/nrf_rpc/entropy_nrf53/cpunet/sysbuild.cmake:9]: How does someone run it then?

Comment (nordicjm) [reply samples/nrf_rpc/entropy_nrf53/cpunet/sysbuild.cmake:9]: Oh the app and net should run for the same image, that is confusing, will update...

Comment (nordicjm) [reply samples/nrf_rpc/entropy_nrf53/cpunet/sysbuild.cmake]: @KAGA164 have reworked this, please check again

Comment (nordicjm) [reply samples/nrf_rpc/entropy_nrf53/cpunet/sysbuild.cmake]: Updated

**Review by KAGA164 — APPROVED** [review]

**Review by peknis — DISMISSED** [review]

**Pull request merged**